build-version.sh

```diff
@@ -40,14 +40,14 @@ else
 fi
 
 BINARIES=$(
-    cat $CRATES2_JSON_PATH | jq -r '
-        .installs | to_entries[] | select(.key|startswith("'${CRATE}' ")) | .value.bins | .[]
-    ' | tr '\r' ' '
+    jq -r '
+        .installs | to_entries[] | select(.key|startswith("'"${CRATE}"' ")) | .value.bins | .[]
+    ' "$CRATES2_JSON_PATH" | tr '\r' ' '
 )
 
-cd $CARGO_BIN_DIR
+cd "$CARGO_BIN_DIR"
 for file in $BINARIES; do
-    if file $file | grep ': data$'; then
+    if file "$file" | grep ': data$'; then
```

Comment:
Could use this instead:

```suggestion
    if [ -f "$file" ] && [ -x "$file" ]; then
```

The first `[ -f "$file" ]` checks to make sure `$file` is a regular file while `[ -x "$file" ]` checks to make sure its executable bit is set.

Reply:
I just realize that this script might need to run on windows, which doesn't have the executable bit, so probably better to go with the original one.

Reply:
the problem that this is trying to catch is that macos runners sometimes don't understand the binaries that had been built (my computer also didn't understand them). The executable bit wouldn't help us here.

I have no idea why this was happening, or whether it still happens, but asking `file` seemed to be enough to catch it.